### PR TITLE
feat: display date as time has passed

### LINKED ISSUE #3653: [feature request] Improving the display of the date
> This ticket is part of series of 7 tickets concluding a design task focusing on improving design and usability of comments feature. Here you can find the [Figma file](https://www.figma.com/board/GfQPMs7PyC7jbkymDTco2D/CP-%2F-Nested-Comments-Improvements?node-id=0-1&t=7sgMA4ZPYXAMb03A-1) for more context.

### **4/7 - Improving the display of the date**
---

**Is your feature request related to a problem? Please describe.**
Currently the date in in a hard-to read format, making it visually cluttered and also it takes time for the user to figure out exactly when was the comment or other content made or edited. Some countries have DD-MM-YYYY and some MM-DD-YYYY formats, so it is not entirely clear which one is month and which one is day.

image

Additionally the text that says '(Edited)' is in a different font and size. In other parts of the platform the solution is better, where the information about editing is incorporated into the date.

image

**Describe the solution you'd like**
Turn date info into what time has passed - 2 hours ago, 5 minutes ago, 1 year ago. Maybe there is some existing library or something that can be implemented, also considering i18n (In english there are only 2 gramatical formulations, such as '1 year ago' and '2 years ago', but in Czech for example you have 'před 1 rokem', 'před 2 roky', 'před 5 lety').

Additionally, it should have a simple [title](https://www.w3schools.com/tags/att_global_title.asp) on hover, like in this example:

image

> Ticket scope to include (notes from Ben):
> 
> - Create component in the library to cover this
> - At least one example in the end-to-end tests of the 'just now' render of the component displayed.
> - List everywhere it should go:
>   - Comments and nested comments across all modules where there are comments
>   - Research module listing
>   - Single research and research updates
>   - Questions module listing
>   - Single question
>   - Single How-To
>   - In academy pages (I am aware that it is a different system, so it is more of a Could-Have, not a Must-Have)

**Describe alternatives you've considered**
An easier alternative would be to just spell out the month into a format like this: 1-Feb-2023, however still a harder to read format.

**Additional context**
See the milestone

### PR DESCRIPTION
**Pull request by onim-at**
## PR Checklist

- [X] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)
- [x] - Unit and e2e tests for the changes that have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix (fixes an issue)
- [X] Feature (adds functionality)
- [ ] Code style update
- [ ] Refactoring (no functional changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?
Dates are shown as timestamp
## What is the new behavior?
Dates are shown as "time has passed"

### To-Do

- [X]   Create component in the library to cover this
- [x]     At least one example in the end-to-end tests of the 'just now' render of the component displayed.
- [x]  Fix ResearchArticle.test.tsx

List everywhere it should go:
- [X]         Comments and nested comments across all modules where there are comments
- [X]         Research module listing
- [X]         Single research and research updates
- [X]         Questions module listing
- [X]         Single question
- [X]         Single How-To 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Git Issues

Closes #3653

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by benfurber** [comment]
Nice one so far @onim-at - looks like exactly the right direction. 🥳

**Comment by onim-at** [comment]
Hello @benfurber! I've added a check on the date in the question cypress test, is this what you intended? 
Also I don't know where to ask so I'll try here, is it possible to join the discord developer chat?

**Comment by benfurber** [comment]
@onim-at Nice one, thank you. I'll have a look today. [You shoule be able to join discord with this link](https://discord.gg/gJ7Yyk4)

File: packages/components/src/DisplayDate/DisplayDate.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const relativeDateFormat = (d: DateType | undefined): string => {
+  if (!d) {
+    return ''
+  }
+  return formatDistanceToNow(new Date(d), { addSuffix: true })
+}
+
+export const DisplayDate = ({ date }: DisplayDateProps) => {
+  console.log(date)
```
Comment (benfurber) [inline packages/components/src/DisplayDate/DisplayDate.tsx:26]: The linter should have caught this. Please drop the console log.

**Review by benfurber — CHANGES_REQUESTED** [review]
Just the one thing that's an absolute must change with the console log but there's a couple of bits I'd appreciate.

If you have a look at `DiscussionTitle` you'll see a style of stories/tests I think would be good. 

The stories file has a separate function for each display expected (roughly) and then a test checks that each function has the expected output. So in this instance roughly somthing like:

stories:
```
export const TwoMonthsAGo: StoryFn<typeof DisplayDate> = () => {
  const twoMonthsAGo = subMonths(new Date(), 2)
  return (
      <DisplayDate date={twoMonthsAGo}></DisplayDate>
)}
```

test: 
```
describe('DisplayDate', () => {
...
  it('renders correctly when two months ago', () => {
    const { getByText } = render(
      <TwoMonthsAGo {...(TwoMonthsAGo.args as IProps)} />,
    )

    expect(getByText("2 months ago")).toBeInTheDocument()
  })
```

Then it's a quick test of our implementation of the package (not the package itself) and more importantly, prevents a regression.

**Comment by dalibormrska** [comment]
Looks absolutely awesome! I'm really happy to see it implemented! Great job @onim-at and @benfurber!  🎉

**Pull request merged**